### ISSUE TITLE
ValueError: invalid version number '0.17.0rc0'

### ISSUE THREAD

**Issue by nagolinc** [huggingface/diffusers#3703]
### Describe the bug

Loading a diffusion model broke suddenly.

I've already tried pip reinstalling diffusers (from pip as well as this github repo)

### Reproduction

from diffusers import DiffusionPipeline
import torch

model_id = "runwayml/stable-diffusion-v1-5"

ref_pipe = DiffusionPipeline.from_pretrained(
            model_id,
            # revision=revision,
            custom_pipeline="stable_diffusion_reference",
            torch_dtype=torch.float16,
            use_auth_token=True,
            cache_dir="./AI/StableDiffusion"
        )

### Logs

```shell
─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Users\Nagol\AppData\Local\Temp\ipykernel_68840\2161590897.py:6 in <module>                    │
│                                                                                                  │
│ [Errno 2] No such file or directory:                                                             │
│ 'C:\\Users\\Nagol\\AppData\\Local\\Temp\\ipykernel_68840\\2161590897.py'                         │
│                                                                                                  │
│ C:\Users\Nagol\AppData\Local\Programs\Python\Python310\lib\site-packages\diffusers\pipelines\pip │
│ eline_utils.py:884 in from_pretrained                                                            │
│                                                                                                  │
│    881 │   │   # 1. Download the checkpoints and configs                                         │
│    882 │   │   # use snapshot download here to get it working from from_pretrained               │
│    883 │   │   if not os.path.isdir(pretrained_model_name_or_path):                              │
│ ❱  884 │   │   │   cached_folder = cls.download(                                                 │
│    885 │   │   │   │   pretrained_model_name_or_path,                                            │
│    886 │   │   │   │   cache_dir=cache_dir,                                                      │
│    887 │   │   │   │   resume_download=resume_download,                                          │
│                                                                                                  │
│ C:\Users\Nagol\AppData\Local\Programs\Python\Python310\lib\site-packages\diffusers\pipelines\pip │
│ eline_utils.py:1265 in download                                                                  │
│                                                                                                  │
│   1262 │   │   │   ]                                                                             │
│   1263 │   │   │                                                                                 │
│   1264 │   │   │   # retrieve passed components that should not be downloaded                    │
│ ❱ 1265 │   │   │   pipeline_class = _get_pipeline_class(                                         │
│   1266 │   │   │   │   cls, config_dict, custom_pipeline=custom_pipeline, cache_dir=cache_dir,   │
│   1267 │   │   │   )                                                                             │
│   1268 │   │   │   expected_components, _ = cls._get_signature_keys(pipeline_class)              │
│                                                                                                  │
│ C:\Users\Nagol\AppData\Local\Programs\Python\Python310\lib\site-packages\diffusers\pipelines\pip │
│ eline_utils.py:336 in _get_pipeline_class                                                        │
│                                                                                                  │
│    333 │   │   else:                                                                             │
│    334 │   │   │   file_name = CUSTOM_PIPELINE_FILE_NAME                                         │
│    335 │   │                                                                                     │
│ ❱  336 │   │   return get_class_from_dynamic_module(                                             │
│    337 │   │   │   custom_pipeline, module_file=file_name, cache_dir=cache_dir, revision=revisi  │
│    338 │   │   )                                                                                 │
│    339                                                                                           │
│                                                                                                  │
│ C:\Users\Nagol\AppData\Local\Programs\Python\Python310\lib\site-packages\diffusers\utils\dynamic │
│ _modules_utils.py:445 in get_class_from_dynamic_module                                           │
│                                                                                                  │
│   442 │   cls = get_class_from_dynamic_module("sgugger/my-bert-model", "modeling.py", "MyBertM   │
│   443 │   """                                                                                 │
│   444 │   # And lastly we get the class inside our newly created module                          │
│ ❱ 445 │   final_module = get_cached_module_file(                                                 │
│   446 │   │   pretrained_model_name_or_path,                                                     │
│   447 │   │   module_file,                                                                       │
│   448 │   │   cache_dir=cache_dir,                                                               │
│                                                                                                  │
│ C:\Users\Nagol\AppData\Local\Programs\Python\Python310\lib\site-packages\diffusers\utils\dynamic │
│ _modules_utils.py:264 in get_cached_module_file                                                  │
│                                                                                                  │
│   261 │   │   resolved_module_file = module_file_or_url                                          │
│   262 │   │   submodule = "local"                                                                │
│   263 │   elif pretrained_model_name_or_path.count("/") == 0:                                    │
│ ❱ 264 │   │   available_versions = get_diffusers_versions()                                      │
│   265 │   │   # cut ".dev0"                                                                      │
│   266 │   │   latest_version = "v" + ".".join(__version__.split(".")[:3])                        │
│   267                                                                                            │
│                                                                                                  │
│ C:\Users\Nagol\AppData\Local\Programs\Python\Python310\lib\site-packages\diffusers\utils\dynamic │
│ _modules_utils.py:46 in get_diffusers_versions                                                   │
│                                                                                                  │
│    43 def get_diffusers_versions():                                                              │
│    44 │   url = "https://pypi.org/pypi/diffusers/json"                                           │
│    45 │   releases = json.loads(request.urlopen(url).read())["releases"].keys()                  │
│ ❱  46 │   return sorted(releases, key=StrictVersion)                                             │
│    47                                                                                            │
│    48                                                                                            │
│    49 def init_hf_modules():                                                                     │
│                                                                                                  │
│ C:\Users\Nagol\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\_distutils\v │
│ ersion.py:54 in __init__                                                                         │
│                                                                                                  │
│    51 │                                                                                          │
│    52 │   def __init__(self, vstring=None):                                                      │
│    53 │   │   if vstring:                                                                        │
│ ❱  54 │   │   │   self.parse(vstring)                                                            │
│    55 │   │   warnings.warn(                                                                     │
│    56 │   │   │   "distutils Version classes are deprecated. "                                   │
│    57 │   │   │   "Use packaging.version instead.",                                              │
│                                                                                                  │
│ C:\Users\Nagol\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\_distutils\v │
│ ersion.py:157 in parse                                                                           │
│                                                                                                  │
│   154 │   def parse(self, vstring):                                                              │
│   155 │   │   match = self.version_re.match(vstring)                                             │
│   156 │   │   if not match:                                                                      │
│ ❱ 157 │   │   │   raise ValueError("invalid version number '%s'" % vstring)                      │
│   158 │   │                                                                                      │
│   159 │   │   (major, minor, patch, prerelease, prerelease_num) = match.group(1, 2, 4, 5, 6)     │
│   160                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ValueError: invalid version number '0.17.0rc0'
```

### System Info

windows+python3.10

**Comment by rishabh063** [comment]
Having the same issue

**Comment by Gustavof1** [comment]
I'm getting this problem when I try to use the controlnet Tile Version:
[lllyasviel/control_v11f1e_sd15_tile](https://huggingface.co/lllyasviel/control_v11f1e_sd15_tile)

It has been removed from the [documentation page](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/controlnet) as well, with no traces of any modification. 

And installing the 0.17.0rc0 diffusers version **don't** fix the problem.
`pip uninstall diffusers`
`pip install diffusers==0.17.0rc0`

**Comment by SomaNonaka** [comment]
It looks like it's fixed... ?

The URL
`https://pypi.org/pypi/diffusers/json`
referenced in
https://github.com/huggingface/diffusers/blob/cd9d0913d90b52d3609427ea2ad5d07b557677dd/src/diffusers/utils/dynamic_modules_utils.py#L43-L49
seems to have been temporarily corrupted.

**Comment by ryancustomuse** [comment]
We are going to try and patch over that file with this updated function

```
def get_diffusers_versions():
    f = open("/app/patch/temp.json")
    releases = json.load(f)["releases"].keys()
    return sorted(releases, key=StrictVersion)
```

**Comment by joshuacc** [comment]
Same issue here using a custom pipeline. It broke sometime in the last 12 hours.

**Comment by ryancustomuse** [comment]
Ahh cool, thank you

**Comment by ghunkins** [comment]
Dealing with the same issue here.

**Comment by KaustubhPatange** [comment]
Same issue here

**Comment by rishabh063** [comment]
> def get_diffusers_versions():
>     f = open("/app/patch/temp.json")
>     releases = json.load(f)["releases"].keys()
>     return sorted(releases, key=StrictVersion)

tried replacing the original function with this still Not working

**Comment by reimager** [comment]
Same issue.
I set local_files_only=True, HF_HUB_OFFLINE=1, etc when running diffusers to avoid issues like this, but this still has brought down all the servers as they can't load pipelines

**Comment by bghira** [comment]
it can't load the one community pipeline, because you're loading it without pinning to a specific revision.

also, using the community pipeline in production is likely not a great idea for this reason.

**Comment by adhikjoshi** [comment]
same issue @patil-suraj @patrickvonplaten ,

Why there is no offline mode? Like, i don't want diffusers to work without making any api call to any remote URL

**Comment by joshuacc** [comment]
For those looking for a quick solution, I ended up downloading the custom model I was using and putting it in a directory next to my app's main python code. No more need for it to be dynamically loaded.

**Comment by vicky-aipaintr** [comment]
> For those looking for a quick solution, I ended up downloading the custom pipeline I was using and putting it in a directory next to my app's main python code. No more need for it to be dynamically loaded.

@joshuacc can you expand more ? I am using lpw_stable_diffusion.py, putting it next to main.py is not working.

**Comment by patrickvonplaten** [comment]
Aiii, the custom pipelines again :-/ We need to test here better! Super sorry, fixing now

**Comment by joshuacc** [comment]
@vicky-aipaintr Sure. Here's what I did.

1. Create a new directory next to my main python code named `lpw_stable_diffusion`.
2. Download the custom pipeline from GitHub and save it in `lpw_stable_diffusion/pipeline.py`
3. Change the custom pipeline argument in my code from `"lpw_stable_diffusion"` to `"./lpw_stable_diffusion/."` so that it knows to load it from the local directory.

This same process should work for other custom pipelines as well.

**Comment by patrickvonplaten** [comment]
Want to explain here a bit what happened :sweat_smile: 

1. We will do a release tomorrow which is why we did a pre-release today so that some bigger UIs can test diffusers: https://github.com/invoke-ai/InvokeAI/pull/3512 . This is good practice when the scope of a project gets bigger
2. The problem is that pre-releases follow a different pattern
![Screenshot from 2023-06-07 18-24-05](https://github.com/huggingface/diffusers/assets/23423619/a74f988f-7067-4699-918e-21a3cc7c2411)

Now when we try to use custom pipelines from GitHub, we always make sure that we people can only download custom pipelines from existing versions and therefore we check on PyPI which versions exist. Sadly did our sorting algorithm break when seeing the new pattern of pre-releases (`0.17.0rc0` as there is no `.` between the last 0 and `rc0`)

This is now fixed in #3708. 
In order to make sure past versions work with community pipelines I sadly have to delete the `0.17.0rc0` version now.

**Comment by vicky-aipaintr** [comment]
Thanks for fixing this @patrickvonplaten. If this is only related to community pipelines maybe the only way to run them should be by downloading them locally as mentioned by @joshuacc .

**Comment by joshuacc** [comment]
I'd actually like to suggest that the community pipelines be loaded from the python package the same way as the diffusers library itself. That's what I was expecting (since I didn't read the docs closely enough).

But I strongly recommend that pipelines shouldn't be downloaded unless the user is explicitly opting into that. For example, by using a method called `download()`. :)

**Comment by KaustubhPatange** [comment]
@vicky-aipaintr You can also keep `lpw_stable_diffusion.py` in your project locally & import `StableDiffusionLongPromptWeightingPipeline` to instantiate it manually via `.from_pretrained(...)` method.

**Comment by bghira** [comment]
> But I strongly recommend that pipelines shouldn't be downloaded unless the user is explicitly opting into that. For example, by using a method called 

that would break it for users that expect it to download every time now.

or users that don't know it downloads every time.

**Comment by reimager** [comment]
> # Everything should work again now as 0.17.0rc0 is deleted.
> 
> Very sorry about this. We therefore won't be able to do pre-releases until a while to keep backwards compatibility with older code versions.

Thanks for fix - you could always do 0.17.0a0 as StrictVersion will parse that fine - it just won't allow "rc"
Not ideal, but would be backwards compat

**Comment by kosukekurimoto** [comment]
```
!pip install diffusers==0.16.1
!pip install accelerate==v0.19.0
!git clone https://github.com/huggingface/diffusers -b v0.16.1 --depth 1 
%cd /content/diffusers/examples/dreambooth
!pip install -r requirements.txt
```
